Added metadata-services module  (issue #197)

marcozampa1970 opened the pull request:
https://github.com/geosolutions-it/csi-sira/issues/197

I positioned the metadata-services module under the frontend folder as this module must be deployed in the sira war.

In pom.xml under the root I added references to the metadata-services module.

In mvc-dispatcher-servlet.xml files under the web module I added the metadata bean (pointing to MetadataController class) entry point of all
requests made to the metadata-services module.

NOTE: In the mvc-dispatcher-servlet.xml files I have changed the version from 3.2 to 3.0 in the attribute "xsi:schemaLocation" because I had some runtime errors.

In the pom.xml of the web form I added the dependency to metadata-services module.



[Metadata services 1.0.pdf](https://github.com/geosolutions-it/csi-sira/files/580588/Metadata.services.1.0.pdf)

mbarto commented:
Cannot merge it right away.

A datasource (siradecsvilDS) configuration is needed. Please, explain how to configure it for our demo server.

marcozampa1970 commented:
Updated issue #197:

Updated jetty-maven-plugin to version 9.3.14 and configured siradecsvilDS datasource to use metadata services

1) Added jetty-env.xml in webapp/WEB-INF

2) In WEB-INF/web.xml added:

	<!-- Datasource used by metadata-services -->
	<resource-ref>
		<res-ref-name>jdbc/siradecsvilDS</res-ref-name>
		<res-type>javax.sql.DataSource</res-type>
		<res-auth>Container</res-auth>
	</resource-ref>

3) In web/pom.xml changed jetty conf in:

			<plugin>
				<groupId>org.eclipse.jetty</groupId>
				<artifactId>jetty-maven-plugin</artifactId>
				<version>9.3.14.v20161028</version>
				<configuration>
					<webApp>
						<contextPath>/sira</contextPath>
					</webApp>
					<httpConnector>
						<port>9191</port>
						<idleTimeout>60000</idleTimeout>
					</httpConnector>
					<war>${project.basedir}/target/sira.war</war>
				</configuration>
				<dependencies>
					<dependency>
						<groupId>c3p0</groupId>
						<artifactId>c3p0</artifactId>
						<version>0.9.1.2</version>
					</dependency>
					<dependency>
						<groupId>postgresql</groupId>
						<artifactId>postgresql</artifactId>
						<version>9.1-901-1.jdbc4</version>
					</dependency>
				</dependencies>
			</plugin>
